### PR TITLE
First batch of type issues from mypy

### PR DESCRIPTION
**Pull request by dstrain115**
- cirq.Qid does not have a name, so changing to NamedQubit in typing.
- Also misc typing fixes all over.
- Remove gateset in engine.sampler, since it no longer exists in Cirq 1.0.

### CODE REVIEW

File: unitary/engine_utils.py [hunk 1]
```diff
@@ -247,13 +247,7 @@ def engine(self):
...
 
     def get_sampler(self, gateset: str = None):
-        if gateset == 'sycamore':
-            gateset = cg.SYC_GATESET
-        elif gateset == 'sqrt-iswap':
-            gateset = cg.SQRT_ISWAP_GATESET
-        else:
-            raise ValueError("Unknown gateset {}".format(gateset))
-        return self.engine.sampler(processor_id=self.processor_id, gate_set=gateset)
+        return self.engine.sampler(processor_id=self.processor_id)
```
Comment (losos0) [inline unitary/engine_utils.py:250]: What's this part? Doesn't seem related to type checking.

File: unitary/quantum_chess/quantum_board.py [hunk 2]
```diff
@@ -135,7 +135,9 @@ def with_state(self, basis_state: int, reset_move_history=True) -> "CirqBoard":
         self.post_selection = {}
         self.circuit = cirq.Circuit()
         self.ancilla_count = 0
-        initial_square_probs = (float(nth_bit_of(i, basis_state)) for i in range(64))
+        initial_square_probs = list(
```
Comment (losos0) [inline unitary/quantum_chess/quantum_board.py:138]: Use a list comprehension

File: unitary/engine_utils.py (hunk 1 above)
Comment (dstrain115) [reply unitary/engine_utils.py:250]: gateset parameter no longer exists in Cirq 1.0.  Added to the PR description.

I can split it out into a different PR if you want.

File: unitary/quantum_chess/quantum_board.py (hunk 2 above)
Comment (dstrain115) [reply unitary/quantum_chess/quantum_board.py]: Done.

File: unitary/engine_utils.py (hunk 1 above)
Comment (losos0) [reply unitary/engine_utils.py:250]: That's all right. A different commit would have been nice though.

File: unitary/quantum_chess/move.py [hunk 3]
```diff
@@ -185,9 +185,9 @@ def to_string(self, include_type=False) -> str:
           include_type: also include the move type/variant in the string
         """
         movestr = self.source + self.target
-        if self.is_split_move():
+        if self.is_split_move() and self.target2:
```
Comment (losos0) [inline unitary/quantum_chess/move.py:188]: This change seems suboptimal since if you had a wrongly constructed split move, then the evidence would be covered up when printing it as a string. I think you could do `str(self.target2)` instead

**Review by losos0 — APPROVED** [review]

File: unitary/quantum_chess/move.py (hunk 3 above)
Comment (dstrain115) [reply unitary/quantum_chess/move.py]: Good idea.  Done.
